x-pack/plugins/apm/server/lib/service_map/get_service_map.ts

```diff
@@ -147,8 +147,12 @@ function getAnomaliesData(options: IEnvOptions) {
     body: {
       size: 0,
       query: {
-        exists: {
-          field: 'bucket_span'
+        bool: {
+          filter: {
+            term: {
+              result_type: 'record'
+            }
+          }
```

Comment:
Two things:

1) shouldn't the query be limited by a couple of things (environment*, date range etc.)? Was this left out intentionally in the previous PR (cc @smith )?

2) What is `record_score` exactly? Is the the record within a time range? Or the highest score ever recorded? We can only use it if it's the highest score within the time range we are querying for

* it doesn't seem like ML jobs take environment into account (only service name and transaction type). This seems like a big deal if i'm not mistaken.

Reply:
1. we can make sure to add a date range filter, but i don't think we can filter by environment currently since it's not retained in the ml index.
2. this is a case where `record_score` seems appropriate to use after reading https://www.elastic.co/blog/machine-learning-anomaly-scoring-elasticsearch-how-it-works last paragraph